### ISSUE TITLE
`has rvm` always returns true

### ISSUE THREAD

**Issue by ziadsawalha** [direnv/direnv#1008]
**Describe the bug**
`has rvm` always returns true because stdlib defines `rvm()`

**To Reproduce**
On a machine without rvm installed, the following  .envrc says ""✅ 'rvm' is detected":
```
if has rvm; then
  echo "✅ 'rvm' is detected"
else
  echo "❌ 'rvm' not detected"
fi
```

**Expected behavior**
has rvm should not return true (or document an alternative method to detect if rvm is available)

**Environment**
 - OS: tested on macOS Catalina, and Alpine
 - Shell: zsh, ash, ..
 - Direnv version 2.32.1

**Additional context**

**Comment by zimbatm** [comment]
Yes, it's unfortunate. I should change the `has()` definition to `type -p "$1" >/dev/null` so it only returns true when there is a command. But it would break back-compat and we don't break back-compa.

The best is to replace the rvm check with `if type -p rvm >/dev/null; then ...`
